pandas/core/arrays/arrow/array.py

```diff
@@ -127,12 +127,34 @@ def floordiv_compat(
         left: pa.ChunkedArray | pa.Array | pa.Scalar,
         right: pa.ChunkedArray | pa.Array | pa.Scalar,
     ) -> pa.ChunkedArray:
-        # Ensure int // int -> int mirroring Python/Numpy behavior
-        # as pc.floor(pc.divide_checked(int, int)) -> float
-        converted_left = cast_for_truediv(left, right)
-        result = pc.floor(pc.divide(converted_left, right))
-        if pa.types.is_integer(left.type) and pa.types.is_integer(right.type):
+        divided = pc.divide(left, right)
```

Comment:
original code used divide, but should it be divide_checked? As all other arithmetic operators use checked version. Currently, (-1 << 63) // -1 silently overflows.

Reply:
The old code went through double, so it did actually overflow when casting back to an int, I've restored the original behavior.